### DISCUSSION TITLE
Getting photon counts

### DISCUSSION THREAD

**Discussion by spangeni** [lumicks/pylake#177] [General]
Is there a way to get blue and red photon counts after tracking the kymolines.For example, if i use tracegreedy then have 4 kymo lines, i wanted to know if i can extract photon counts for each of the kymolines apart from position and coordinates.To be specific, I wanted to have a way to get the photon counts of specific kymolines from kymograph that has many kymolines.Any suggestions would be helpful.

**Comment by JoepVanlier** [comment]
Hi, 

sorry, I didn't see your question until today.

You can sample the photon counts by iterating over the Kymoline lists. See the following for an example: https://lumicks-pylake.readthedocs.io/en/latest/tutorial/kymotracking.html#extracting-summed-intensities

For a single trace, it looks something like this:
```python
longest_trace.sample_from_image(num_pixels=5)
```
This will sum the counts for each point on that trace in 5 pixels in either direction (from `position - num_pixels` to (and including) `position + num_pixels`).

If you want to do this for all your kymolines, you can iterate over them.
```python
sampled_counts = []
for line in kymowidget.lines:
    sampled_counts.append(line.sample_from_image(num_pixels=5))

print(sampled_counts)
```

We can also write that a bit shorter using a list comprehension:
```python
sampled_counts = [line.sample_from_image(num_pixels=5) for line in kymowidget.lines]
```

I hope I'm interpreting your question correctly. If the question is more related to identifying which `Kymoline` is which then I'm afraid we don't have specific functionality for that yet. What you could do is select your line of interest by dragging a region to track using the right mouse button, rather than using `track all` button.

Maybe you can explain a little more about your use case. What are you trying to achieve?
